### DISCUSSION TITLE
Azure IOT

### DISCUSSION THREAD

**Discussion by michelkamp** [Koenkk/zigbee2mqtt#7660] [General]
Hi,

Just a short question. Will this also work with an Azure IOT HUB ?

I tried but can't get it to work.

1. created the IOT HUB
2. created the device
3. for the host i used MQTTS on 8883 mqtts://{your_iothub_name}
4. constructed the user : {your_iothub_name}.azure-devices.net/{your_device_id}/?api-version=2018-06-30
5. constructed the password using the generated the SAS  
6. and set the force_disbale_reain to true

but still i get a error "not connected to MQTT server!"

Any help would be welcome

Michel

**Comment by michelkamp** [comment]
@Koenkk any idea on this ? Thanks

**Reply by Koenkk** [reply]
I don't have any experience with Azure IOT, @chrisns if I understood correctly you made z2m run in the cloud (https://github.com/Koenkk/zigbee2mqtt/pull/4948). Do you know what is wrong?

**Reply by chrisns** [reply]
¯\_(ツ)_/¯
is anything publishing with a retain?
any debug stuff from mqtt available to see the connection, does it get connected and then booted etc?
I only did aws-iot but i presume azure is similar

**Reply by michelkamp** [reply]
Hi , Thanks for the reply. I enabled debugging . Before the connect i only get a debug message "using MQTT login with username: xxxxxx" and the next message is the 'not connected to MQTT server' and after this no debug message related to the connect.

I did set force_disable_retain to true and also version: to 3 , but i guess it should connect first ?

Would be great to see this also working with Azure IOT. So if i can help you out here just let me know .. i can provide you access to my HUB or help you testing...

**Reply by jihadelsayed** [reply]
Hello @michelkamp !
Did you success in connecting to azure IOT or did you have another way!
if you success can you please share it with me.

**Comment by michelkamp** [comment]
Hi,

I didn’t try it anymore. So could be that’s now working. As workaround i solved  it by using the Log analytics API . As long I had the data in azure I was fine .

Michel


This email is typed on my mobile phone and so can contain typo’s.
________________________________

**Reply by jihadelsayed** [reply]
Thank you for the reply!
I think i will try Azure Device Provisioning Service but if it didn't work i will go with Log analytics API.
